src/Microsoft.AspNetCore.Blazor.Server/Circuits/RemoteRenderer.cs

```diff
@@ -85,10 +94,48 @@ public void Dispose()
         }
 
         /// <inheritdoc />
-        protected override void UpdateDisplay(in RenderBatch batch)
+        protected override Task UpdateDisplay(in RenderBatch batch)
         {
-            var task = _client.SendAsync("JS.RenderBatch", _id, batch);
-            CaptureAsyncExceptions(task);
+            // Prepare to track the render process with a timeout
+            var renderId = Interlocked.Increment(ref _nextRenderId);
+            var pendingRenderInfo = new AutoCancelTaskCompletionSource<object>(TimeoutMilliseconds);
+            _pendingRenders[renderId] = pendingRenderInfo;
+
+            // Send the render batch to the client
+            // If the "send" operation fails, abort the whole render with that exception
+            _client.SendAsync("JS.RenderBatch", _id, renderId, batch).ContinueWith(sendTask =>
```

Comment:
can `_client.SendAsync` throw synchronously? We'd leak in that case.

Reply:
Good point - I'll catch that.